operators/pkg/instance-controller/container_logic.go

```diff
@@ -256,6 +278,24 @@ func (r *InstanceReconciler) CreateContainerEnvironment(
 		return err
 	}
 
+	if !environment.Resources.Disk.IsZero() {
+		pvc := v1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+			},
+			Spec: buildContainerInstancePVCSpec(environment),
+		}
+
+		if _, err := ctrl.CreateOrUpdate(ctx, r.Client, &pvc, func() error {
+			// PVC's specs are immutable, they have to be set directly
```

Comment:
Even though they are immutable, the correct way would be set them here, checking if the creationTimeStamp is equal to zero (as in the example).

Reply:
I tried but if I set the specs inside the CreateOrUpdate it fails saying they're immutable...